### DISCUSSION TITLE
How to get more engagements on my github repository ?

### DISCUSSION THREAD

**Discussion by MOHAMED-EHAB-DEV** [community/community#18920] [Other Feature Feedback, Questions, & Ideas]
How to get more engagements on my github repository ?

**Comment by MathiasReker** [comment]
You can blog on medium.com

**Reply by MOHAMED-EHAB-DEV** [reply]
I share it on all the social media platforms but i don't get anything

**Reply by d-coder111** [reply]
Same thing I want to know

**Comment by Tran-Quyen** [comment]
You can share your repo to social. It help everyone know your contribute to Github and IT

**Comment by KrishChopra22** [comment]
**Learning in Public is the most important thing!**
Whatever projects you are working on, simply share the link of that repo and alongwith your accomplishment on social handles like Twitter & tag your @community-friends. 
*There are various hashtags which are Trending like*
**#100DaysOfCode**
**#OpenSource**
**#OpenSourceProjects**

Doing these things will automatically gain you reach but also you will come in the eyes of Industry Leaders that will definitely help you in future, building your Career ✨

**Comment by Zi-Tao** [comment]
@Programming-School-Pro-Coding https://www.freecodecamp.org/news/increase-engagement-on-your-public-github-repositories/

**Comment by rezabintami** [comment]
@Programming-School-Pro-Coding 
you can share your repo on social media or something else. help contribute to any others people and you can make tutorials in medium

**Comment by radar-blip** [comment]
You could try adding more topics (keywords) to each of your repositories descriptions. This will make it easier for others searching for same topics or interests to find your repositories.

**Comment by edward795** [comment]
**These are some of the things u might want to do:**

1. *Publish cool Readme to make the repo worth exploring
2. *Make repos for projects that are unique 
3. *Engage more in community discussions to get your profile noticed.
4. *No mysteries, always give tips & comments in your code  so that other fellow programmers can follow.

**Comment by mukeshsolanki** [comment]
I have a few decent repositories with decent stars. Here's how I get traction with these repos.

**On Github**
1. Make sure the repository is well maintained.
2. Documentation is key make sure the readme file has details required to use your lib, SDK etc... and yeah make sure also throw in screenshots to show off any UI if your repo provides them.
3. Use badges. This will help identify the env and the min, max and current version of the repo.

**External Sources**
1. Write a blog post about the problem at hand and how the repo, lib, SDK etc... solves the issue.
2. Share them on linked in, twitter, and other socials. (I usually don't share the repo instead I share the blog post which makes for better SEO so google picks it up faster :P )

**Comment by OliverHarris786** [comment]
To get more engagement on your GitHub repository, start by writing clear documentation and making your project easy to understand. Share it on social media, relevant forums, and communities like [Baddies Hub](https://baddiehub.news/) to get more visibility. Collaborating with others and contributing to their repos can also help bring attention to yours!

**Comment by adnanmalik83** [comment]
Please check if my repository is presentable https://github.com/adnanmalik83?tab=repositories
